### DISCUSSION TITLE
Flash programmer for Linux

### DISCUSSION THREAD

**Discussion by bobbarnes1981** [slu4coder/Minimal-64x4-Home-Computer#13] [Show and tell]
I assembled and used this programmer (https://github.com/slu4coder/SST39SF010-FLASH-Programmer) to program my flash ICs for the Minimal64x4.

Here is my port of the prom.exe to python that I used to program my flash under linux (https://github.com/bobbarnes1981/SST39SF010-FLASH-Programmer/tree/main/PromPy). It should be a reasonably faithful conversion of the C++ code to python.

**Comment by slu4coder** [comment]
Thanks, Bob, for making this platform-independent.

**Reply by slu4coder** [reply]
I encountered a minor issue with this on Windows. Windows requires the port as a plain string like "COM9". For Linux, you have added a "/dev/" infront of that. Here is my suggestion:

import platform

def getFirstComPort():
    for port in serial.tools.list_ports.comports():
        if platform.system() == "Linux": return "/dev/" + port.name
        else: return port.name
    return None

Otherwise, this is working just fine ;-) For cosmetics, I'd suggest to replace "[G" with nothing.

**Reply by bobbarnes1981** [reply]
Thanks. I have added that in.
